app/models/hyrax/permission_template.rb

```diff
@@ -22,6 +22,19 @@ def agent_ids_for(agent_type:, access:)
       access_grants.where(agent_type: agent_type, access: access).pluck(:agent_id)
     end
 
+    # @api public
+    #
+    # Create a permission access if it doesn't already exist
+    # @param agent_type [String] agent type (e.g. Hyrax::PermissionTemplateAccess::GROUP, Hyrax::PermissionTemplateAccess::USER)
+    # @param agent_id [String] user id or group name who has access
+    # @param access [String] type of access (e.g. Hyrax::PermissionTemplateAccess::MANAGE, Hyrax::PermissionTemplateAccess::DEPOSIT, Hyrax::PermissionTemplateAccess::VIEW)
+    # @return [Hyrax::PermissionTemplateAccess] the created or found access
+    def find_or_create_access(agent_type:, agent_id:, access:)
+      records = Hyrax::PermissionTemplateAccess.where(permission_template_id: id, agent_type: agent_type, agent_id: agent_id, access: access)
+      return records.first if records.present? # should only be one match
+      access_grants.create(agent_type: agent_type, agent_id: agent_id, access: access)
+    end
+
     # The list of workflows that could be activated; It includes the active workflow
```

Comment:
Allows migration service to try and grant admin's edit access without having to check if it already exists.  If you just try to add it, a not-uniq exception is raised.

Reply:
Nice.  I switched to use the standard method.